app/uk/gov/hmrc/cgtpropertydisposalsfrontend/controllers/returns/triage/MultipleDisposalsTriageController.scala

```diff
@@ -927,24 +928,72 @@ class MultipleDisposalsTriageController @Inject() (
+                                               .copy(
+                                                 taxYear = existingTaxYear,
+                                                 completionDate = Some(CompletionDate(shareDisposalDate.value)),
+                                                 taxYearAfter6April2020 = Some(existingTaxYear.isDefined)
+                                               )
+                                           )
+                                         )
+                      newState         = updateState(
+                                           state,
+                                           updatedAnswers,
+                                           d =>
+                                             d.bimap(
+                                               multipleIndirect =>
+                                                 multipleIndirect.copy(
+                                                   exampleCompanyDetailsAnswers = multipleIndirect.exampleCompanyDetailsAnswers,
+                                                   yearToDateLiabilityAnswers = None,
+                                                   gainOrLossAfterReliefs = None
+                                                 ),
+                                               multiple =>
+                                                 multiple.copy(
+                                                   examplePropertyDetailsAnswers = multiple.examplePropertyDetailsAnswers
+                                                     .map(_.unset(_.disposalDate)),
+                                                   yearToDateLiabilityAnswers = None,
+                                                   gainOrLossAfterReliefs = None
+                                                 )
+                                             )
+                                         )
+                      _               <- EitherT(
+                                           updateSession(sessionStore, request)(
+                                             _.copy(journeyStatus = Some(newState.merge))
+                                           )
+                                         )
+                    } yield existingTaxYear
+
+                    result.fold(
+                      { e =>
+                        logger.warn("Could not find tax year or update session", e)
+                        errorHandler.errorResult()
+                      },
+                      existingTaxYear =>
+                        if (!isValidTaxYear(existingTaxYear, shareDisposalDate.value))
+                          Redirect(routes.CommonTriageQuestionsController.amendReturnDisposalDateDifferentTaxYear())
+                        else
+                          Redirect(routes.MultipleDisposalsTriageController.checkYourAnswers())
+                    )
+                  case _                                                                    =>
```

Comment:
I'm not sure that you need two cases in the pattern match when it's an amend return and when it's not - they'll both look very similar. The logic with the existing tax year is still wrong as well. Everything could be the same until the right of the fold of the handling of the result - all that needs to go in there is something like:
```scala
if(state.exists(_.originalReturn.disposalDate =!= taxYear))) Redirect(newExitPage)
else Redirect(cya)
```
The original return data for an amend return is in `FillingOutReturn` under a new field  in it called `originalReturn` - that field is defined only if it's an amend return
`.disposalDate` doesn't exist above, but you might find it useful to create it cos you'll need it in multiple places